boards: bluerobotics: navigator: Add initial support

joaomariolago opened the pull request:
Add initial support for Blue Robotics `Navigator Flight Controller` board.

### Solved Problem
Currently there is no support for the Blue Robotics `Navigator Flight Controller` board.

### Solution
- Add initial support for Blue Robotics `Navigator Flight Controller` board.

### Changelog Entry
For release notes:
- Feature:  Add initial support for the Blue Robotics [Navigator Flight Controller](https://bluerobotics.com/store/comm-control-power/control/navigator/) board

### Alternatives
Do not support `Navigator Flight Controller` board.

### Test coverage
- Initial tests were made in real hardware using a raspberry PI 4 with [BlueOS](https://blueos.cloud/)
- Calibration of sensors were made using QGC

### Context
[Navigator Flight Controller](https://bluerobotics.com/store/comm-control-power/control/navigator/)
[Navigator Hardware Setup](https://bluerobotics.com/learn/navigator-hardware-setup/)
[Navigator Schematic](https://bluerobotics.com/wp-content/uploads/2022/06/NAVIGATOR-PCB-SCHEMATIC.pdf)

dagar commented on src/drivers/adc/ads1115/ADS1115.cpp:
```diff
@@ -70,6 +70,11 @@ int ADS1115::probe()
 		return ret;
 	}
 
+	// In flight controllers that operates with multiple autopilots the initial state may be unknown.
```
Alternatively we could tolerate checking for other known configurations.

JoaoMario109 replied:
I am not entirely sure about this one, at least from what I know, the [ADS1115](https://www.ti.com/lit/ds/symlink/ads1115.pdf) doesn't have a `Who Am I` or similar register to verify its identity. Additionally, the CONFIG register have a broad range of values and could be altered by other processes, especially on a Linux board when PX4 isn’t running. This makes it difficult to reliably determine what will be stored in the config register when PX4 is trying to connect.

JoaoMario109 replied:
Hi @dagar, just following up on the review regarding the ADS1115 configuration. I'd would like to know your thoughts when you have a moment. Thanks!

dagar commented on posix-configs/rpi/navigator/sub.config:
```diff
@@ -0,0 +1,94 @@
+#!/bin/sh
+. px4-alias.sh
```
We should really unify this with init with SITL and NuttX.

dagar commented on boards/bluerobotics/navigator/init/rc.board_defaults:
```diff
@@ -0,0 +1,7 @@
+#------------------------------------------------------------------------------
+
+# 1K + 4.7K
+param set-default BAT1_V_DIV 5.7
```
With the way init is currently handled for these linux boards (posix-configs/rpi/navigator/sub.config) this isn't going to be called by default.

If you wanted to work towards that I'd suggest manually calling it yourself, and you could also add an rc.board_sensors.

mrpollo commented:
FYI, I rebased this branch so it can pickup the latest CI changes and we can verify everything works as intended.

patrickelectric commented:
@mrpollo @dagar it appears that the CI issues are unrelated to the PR.

patrickelectric commented:
Can we merge it ? Is there anything holding this PR ?

joaomariolago commented:
Rebased the branch onto main and removed the commit that allowed bypassing the register check for the ADS1115 ADC from this PR, as this issue was addressed by #24428. Including the old bypass check commit would cause unnecessary conflicts. This PR now solely contains the initial support for the Bluerobotics Navigator board.

dakejahl reviewed (approved):
It would be cool to have this in for 1.16, I think there is no harm in merging this since it doesn't impact anything else.

Approving, but I'll let @dagar give it another review before merge.

patrickelectric commented:
:shipit:

slgrobotics commented:
Wouldn't it be prudent to start including **arm64** support in Raspberry Pi based boards (also emlid, scumaker)?

Example here: https://github.com/slgrobotics/PX4-Autopilot/blob/main/boards/emlid/navio2/arm64.px4board

Build command, for example: `make emlid_navio2_arm64`

Pedro-Roque commented:
Hi @joaomariolago , great work with this! I wonder if you plan to / did include support for reading the leak sensors and add failsafes on PX4 to handle them?

joaomariolago commented:
> Hi @joaomariolago , great work with this! I wonder if you plan to / did include support for reading the leak sensors and add failsafes on PX4 to handle them?

Thanks for the feedback! These are on my plans. I don’t have a specific timeline for when I’ll implement them, but I’ve opened #25509 to keep it in track.
